platforms/documentation/docs/src/docs/userguide/authoring-builds/build_environment.adoc

```diff
@@ -131,54 +129,77 @@ spring-boot.version = 2.2.1.RELEASE
-      dryRun = true
-  }
-}
-----
+Project properties are specific to your Gradle project, they can be used to customize your build.
+Project properties can be accessed in your build files and get passed in from an external source when your build is executed.
+Project properties can be retrieved lazily using `providers.gradleProperty()`.
 
 === Setting a project property
 
-You have three options to add project properties, listed in order of priority:
+You have four options to add project properties, listed in order of priority:
 
 1. *Command Line:* You can add project properties directly to your link:{groovyDslPath}/org.gradle.api.Project.html[Project] object via the `-P` command line option.
 +
 [source,text]
 ----
-$ ./gradlew test -Pgradle.testselection.dryRun
+$ ./gradlew build -PmyProperty='Hi, world'
 ----
-2. *As a System Property:* Gradle creates specially-named system properties for project properties which you can set using the `-D` command line flag or `gradle.properties` file. For the project property `prop`, the system property created is called `org.gradle.project.__prop__`.
+2. *System Property:* Gradle creates specially-named system properties for project properties which you can set using the `-D` command line flag or `gradle.properties` file. For the project property `myProperty`, the system property created is called `org.gradle.project.__prop__`.
 +
 [source,text]
 ----
-$ ./gradlew build -Dorg.gradle.project.prop='Hi, world'
+$ ./gradlew build -Dorg.gradle.project.myProperty='Hi, world'
 ----
 +
 .gradle.properties
 [source,properties]
 ----
-org.gradle.project.prop='Hi, world'
+org.gradle.project.myProperty='Hi, world'
 ----
-3. *Environment Variables:* You can set project properties with environment variables. If the environment variable name looks like `ORG_GRADLE_PROJECT___prop__=somevalue`, then Gradle will set a `prop` property on your project object, with the value of `somevalue`.
+3. *Gradle Properties File:* You can also set project properties in `gradle.properties` files.
 +
+.gradle.properties
 [source,text]
 ----
-$ export ORG_GRADLE_PROJECT_myProperty=somevalue
+myProperty='Hi, world'
+----
+4. *Environment Variables:* You can set project properties with environment variables. If the environment variable name looks like `ORG_GRADLE_PROJECT___prop__=somevalue`, then Gradle will set a `prop` property on your project object, with the value of `somevalue`.
++
+[source,text]
+----
+$ export ORG_GRADLE_PROJECT_myProperty='Hi, world'
 ----
 +
 This feature is very useful when you don't have admin rights to a continuous integration server and need to set property values that are not easily visible.
 Since you cannot use the `-P` option in that scenario nor change the system-level configuration files, the correct strategy is to change the configuration of your continuous integration build job, adding an environment variable setting that matches an expected pattern.
 This won't be visible to normal users on the system.
```

Comment:
This example is strange because if you don't have admin rights on a CI server, and are unable to add `-P` to command line invocations, then you most likely don't have permission to add environment variables either. I would even say that typically admins are the _only_ ones that can define environment variables. For example, I can't set environment variables in [gradle/gradle](https://github.com/gradle/gradle), an admin has to do that, but I could create a PR to modify [a workflow file's command line invocation](https://github.com/gradle/gradle/blob/656ada7e12aad698cf1eb6f78e53ac48dcd01b8c/.github/workflows/contributor-pr.yml#L54) as a developer.

The sentiment that using environment variables for CI servers is good to have included in the docs, but I would drop the whole "admin CI server" example and just add something like this to the above list item:


> 4. *Environment Variables:* You can set project properties with [...]. This is typically the preferred way to supply project properties, especially secrets, to unattended builds like those running on CI servers.

Reply:
Sounds good. Added.